docs/adr/0007-azure-pipeline-security-model.md

```diff
@@ -0,0 +1,70 @@
+# Azure pipeline security model
+
+## Status
+
+Proposed
+
+
+## Context
+
+This ADR defines an approach for a least-privilege Azure DevOps Pipeline that supports the automated deployment, teardown and re-deployment of a complete Marain instance.
+
+The challenge here is to balance the varied benefits of having a self-sufficient, repeatable pipeline with the need for adequate security controls to minimise the risk of the pipeline becoming the source of a security incident.
+
+Such a deployment pipeline needs to be able to perform the following actions:
+
+1. Create resources in the target subscription
+1. Query AAD for existing applications & service principals
```

Comment:
I would argue that this is in conflict with the "least privilege" requirement. It is entirely possible to set things up so that the pipeline does not have this capability, because the creation of apps and service principals for inbound auth can be done as a one-time operation by a domain admin, and the relevant IDs then recorded and made available to the pipeline.

It is certainly a good deal more convenient for the pipeline to be able to query AAD, but that's an example of the usual conflict between security and convenience. If we have the pipeline querying AAD then we have made a decision _not_ to apply a "least privilege" rule.

Reply:
Security is always a trade-off with usability.  I think the ADR needs to make a better case for why we believe the trade-off is worth making.

Reply:
We used to provide a separate script for that kind of one time setup. Clients often get picky about this.